packages/nimble-components/src/tooltip/specs/README.md

```diff
@@ -14,6 +14,102 @@ The nimble-tooltip project will first be implemented as a prototype, open issues
 
 ---
 
+
+## Potential approaches
+
+Below we enumerate several different design approaches from which we may choose one or more as our ideal API surface.
+
+### Prior art
+- [Angular Material Tooltip](https://material.angular.io/components/tooltip/overview) - Uses Angular specific functionality (Directive)
+- [Carbon Design System](https://carbondesignsystem.com/components/tooltip/usage#live-demo) - Tooltip component takes anchor element as content.
+- [Atlassian Design System](https://atlassian.design/components/tooltip/examples) - Tooltip component takes anchor element as content
+- [Adobe Spectrum (React)](https://react-spectrum.adobe.com/react-spectrum/Tooltip.html) - Separate wrapper component around tooltip and anchor elements.
+- [FAST](https://explore.fast.design/components/fast-tooltip) - Tooltip associated with anchor element by id.
+
+### Associate by `id`
+This is the FAST approach, and the current implementation. The tooltip has an `anchor` attribute which takes the `id` of the anchor element.
+```html
+<nimble-tooltip anchor="previewButton">...</nimble-tooltip>
+<nimble-button id="previewButton">Preview</nimble-button>
+```
+
+**Pros**
+- No additional implementation cost
+- Familiar pattern, e.g. `<label>` element with `for` attribute, `aria-labelledby` attribute, `aria-describedby` attribute
+- Reasonable DOM structure, i.e. tooltip sibling of anchor element
+
+**Cons**
+- Annoying to have to give anchor element an `id`
+- <span style="color:red">**Not conducive to use in shared components because `id` values must be unique**</span>
+
+### Tooltip slotted in anchor element
+The anchor element defines a `slot` for the tooltip and takes the tooltip as content.
+```html
+<nimble-button>
+    Preview
+    <nimble-tooltip slot="tooltip">...</nimble-tooltip>
+</nimble-button>
+```
+
+**Pros**
+- Familiar slotting pattern
+- Logical DOM structure (element contains its tooltip)
+- May not need `nimble-tooltip` element at all -- could slot tooltip content directly without wrapping in `nimble-tooltip` element
+
+**Cons**
+- Expensive to implement and maintain. <span style="color:red">**Requires forking and modifying the template of every component that wants to support a tooltip.**</span> (Though code duplication could be avoided by using shared template fragments.) Tooltip implementation leaks into every other component.
```

Comment:
As we discussed we have added slots to elements without forking templates, for examples the text field:
https://github.com/ni/nimble/blob/b39cac6d5651afee85bd9f5e18c91d2e7ad17503/packages/nimble-components/src/text-field/index.ts#L59-L68

So I'm not sold on the assertion that every template must be forked for us to add new slots to our components, even today.

> Tooltip implementation leaks into every other component.

I'm not exactly sure what are the cons this statement is trying to express. We call having a start slot that supports an icon a feature not a leak of icon implementation into components. Is there some concern this is expressing?

```suggestion
- Requires per element modification to add new tooltip slots. While we have added new slots to existing elements, there may be situations where it is required to fork a template to support adding a tooltip slot. The exact cost would need prototyping / investigation.
```

Reply:
Based on the other discussion I'm actually a bit more convinced that while still per-component work, we could implement this pattern today with just a slightly less optimal pattern by dynamically inserting slots into templates.